### PR TITLE
Keep Our Light Of Liberty Shining - The TCAF Corvette

### PR DESCRIPTION
**Pull request by RustingWithYou**
Adds a Republican Fleet patrol corvette, finally making use of all @NobleRow's excellent TCAF drip. This vessel can spawn in Tau Ceti, the CRZ, Valley Hale, or the Badlands, and is mutually exclusive with the TCFL ship.

Also adds TCAF gear crates to the antag uplink, as well as a new burst-fire blaster rifle commonly used by Dorn's most expendable soldiers. (sprited by @stev-the-third)

### CODE REVIEW

**Comment by RustingWithYou** [comment]
!review

File: code/game/objects/structures/crates_lockers/crates/gear_loadout.dm [hunk 1]
```diff
@@ -938,6 +938,85 @@
...
+	new /obj/item/shield/energy/legion(src)
+	new /obj/item/gun/energy/blaster/revolver(src)
+	new /obj/item/clothing/accessory/holster/hip(src)
+	new /obj/item/clothing/suit/armor/carrier/tcaf(src)
+	new /obj/item/clothing/head/helmet/tcaf(src)
+	new /obj/item/gun/energy/blaster/tcaf(src)
+	new /obj/item/storage/belt/military(src)
+
+
+
```
Comment (Geevies) [inline code/game/objects/structures/crates_lockers/crates/gear_loadout.dm:1019]:
```suggestion
```

File: code/modules/projectiles/guns/energy/blaster.dm [hunk 2]
```diff
@@ -109,3 +109,30 @@
 		toggle_scope(2.0, usr)
 	else
 		to_chat(usr, "<span class='warning'>You can't look through the scope without stabilizing the rifle!</span>")
+
+/obj/item/gun/energy/blaster/tcaf
+	name = "blaster rifle"
+	desc = "Developed by Zavodskoi Interstellar, the Z.I. Guardian is a burst-fire blaster rifle designed for the armed forces of the Republic of Biesel. Cheap, reliable and easy to mass-produce, these waepons can be seen across the branches of the TCAF."
```
Comment (Geevies) [inline code/modules/projectiles/guns/energy/blaster.dm:115]:
```suggestion
	desc = "Developed by Zavodskoi Interstellar, the Z.I. Guardian is a burst-fire blaster rifle designed for the armed forces of the Republic of Biesel. Cheap, reliable and easy to mass-produce, these weapons can be seen across the branches of the TCAF."
```

**Review by Geevies — CHANGES_REQUESTED** [review]

**Review by DreamySkrell — CHANGES_REQUESTED** [review]
image

since mapdiffbot2 is currently dead...

assuming this is the up to date screenshot posted on maincord:
- make the airlock marked with red into a docking airlock
- add an external access airlock at the front of the ship - I think would be easiest to fit at the area marked with orange
- remove the exterior doors/shutters on the shuttle, marked with yellow (reasoning is that they are easy to forget to close when undocking, leading to venting the shuttle)
- add a docking airlock at the green area, should be easy to fit by moving the shuttle one tile over in either direction (ping me on discord if not sure how to do it)

**Comment by RustingWithYou** [comment]
> image
> 
> since mapdiffbot2 is currently dead...
> 
> assuming this is the up to date screenshot posted on maincord:
> 
> * make the airlock marked with red into a docking airlock
> * add an external access airlock at the front of the ship - I think would be easiest to fit at the area marked with orange
> * remove the exterior doors/shutters on the shuttle, marked with yellow (reasoning is that they are easy to forget to close when undocking, leading to venting the shuttle)
> * add a docking airlock at the green area, should be easy to fit by moving the shuttle one tile over in either direction (ping me on discord if not sure how to do it)

addressed

**Review by Geevies — APPROVED** [review]
code lgtm

**Review by DreamySkrell — APPROVED** [review]

**Review by shaky-jake — CHANGES_REQUESTED** [review]
First things first, air alarms are missing in the port thruster, all four sections of the holding cells area, and the docking port access next to them. You also need emergency shutters. Window spawners handle this automatically of course, but all the reinforced cockpit windows you're using need them to be manually placed. All the doors are missing emergency shutters under them too.

In the same line of thought - Since this is a military ship, exterior windows should really have blast doors to seal them off during combat. Preferably lessen up on them too, 60% of the front facing side of the ship is windows currently.

There's a few piping/wiring issues:
- Missing wire connection between the SMES and the hangar APC.
- There's a bent pipe instead of a straight pipe that cuts the fuel line off to the port thruster. Located just in front of the shuttle's sensor suite.
- Scrubber line for ship is hooked up to CO2 canisters through universal adapter. Same goes for the shuttle. 

As for the shuttle, it's missing an aft targeting entry point. And it can't dock properly. From looking over it briefly, I noticed it's missing interior and exterior sensors. Also, the outside vent should have an ext marker, not an out marker. Reference other shuttles that have functioning automatic docking and mess around with it until it works in a test server.

There's some minor optional stuff too - One being a missing grate over the SMES's terminal. Also, while the shortened railings you used by the grauwolf look nice in StrongDMM, those are unwrenched variants and show up like this in game.
grauwolf_railing_issue

Last couple things would be one, to bump the shuttle transit marker north by 3-4 tiles as currently during transit there's a sudden cutoff to black from the skybox a few tiles out from the cockpit. Two, fix the portside docking port interior access button to not be floating in the air.

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**